src/services/message-relayer.service.ts

```diff
@@ -374,11 +373,24 @@ export class MessageRelayerService extends BaseService<MessageRelayerOptions> {
           .replace(/^0+/, ''),
     ])
 
-    // TODO: Complain if the batch doesn't exist.
+    if (proof === undefined) {
+      this.logger.info('_getMessageProof:: eth_getProof did not return proof.')
+      return // pointless to keep going so return
```

Comment:
Could we actually throw an error here instead? I think that'd be more correct than returning. You could use the same error message.

Reply:
`throw new Error(`_getMessageProof:: eth_getProof did not return proof.`)`

Yes, will do!